### PR TITLE
Added more get functions to PubSubSubscriptionMessage

### PR DESCRIPTION
**Pull request by lclc98**

### CODE REVIEW

File: packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts [hunk 1]
```diff
@@ -4,20 +4,22 @@ import TwitchClient from 'twitch';
...
+	'streak-months': number;
 }
 
 export interface PubSubSubscriptionGiftDetail {
-	context: 'subgift';
+	context: 'subgift' | 'anonsubgift';
 	recipient_id: string;
 	recipient_user_name: string;
 	recipient_display_name: string;
+	months: number;
```
Comment (d-fischer) [inline packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts]: Does this property _always_ exist?

File: packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts [hunk 2]
```diff
@@ -32,26 +34,90 @@ export default class PubSubSubscriptionMessage {
 		this._twitchClient = twitchClient;
 	}
 
-	// these following three methods can't use isGift because it messes up type inference
 	/**
 	 * The ID of the user subscribing to the channel.
 	 */
 	get userId() {
-		return this._data.context === 'subgift' ? this._data.recipient_id : this._data.user_id;
+		return this.isGift || this.isAnonymous
```
Comment (d-fischer) [inline packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts]: Please don't use the other getters internally - then you don't have to use type assertions either.

File: packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts [hunk 3]
```diff
@@ -32,26 +34,90 @@ export default class PubSubSubscriptionMessage {
...
+	 * The cumulative amount of months the user has been subscribed for.
+	 *
+	 * Returns the months if a gift sub or the cumulative months.
+	 */
+	get months() {
+		return this.cumulativeMonths;
+	}
+
+	/**
+	 * The time the user subscribed
```
Comment (d-fischer) [inline packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts]: Please end the descriptions with periods consistently.

File: packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts (hunk 1 above)
Comment (lclc98) [reply packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts]: From looking at the docs it always exists for gift subs, but I will ask in the discord.

File: packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts [hunk 4]
```diff
@@ -32,26 +34,92 @@ export default class PubSubSubscriptionMessage {
...
+	 * The time the user subscribed.
+	 */
+	get time() {
+		return new Date(this._data.time);
+	}
+
+	/**
+	 * The message sent with the subscription
+	 *
+	 * Returns null if the subscription is a gift subscription
```
Comment (d-fischer) [inline packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts]: Periods missing here too

File: packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts (hunk 1 above)
Comment (lclc98) [reply packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts]: From the discord I got told it should be there.

**Pull request merged**